backend/config/corpora-api.yml

```diff
@@ -515,6 +453,54 @@ paths:
         "404":
           $ref: "#/components/responses/404"
 
+  /dp/v1/collections/{collection_uuid}/upload/link:
+    post:
+      summary: Start an dataset upload
+      tags:
+        - collections
+      description: >-
+        An authenticated user can upload a file from a shared link to a dataset in their collection.
+
+        If the upload is in an error state, this endpoint can be used with the dataset_id to restart the collection.
+        If a new link is provided, the new link will be used. If the collection is not in an error state, this endpoint
```

Comment:
nit:

> If the collection is not in an error state, this endpoint will return an error.

I'm trying to unpack what this means 😆 

So if the collection is **not** in an error state, this endpoint will return an error (400, 401, 409, 413)
But if the collection is in an error state, what will the endpoint return?

Reply:
The upload must be in an error state. If it's not in an error state, 409 is the correct response to return, but for now you can't retry an upload. All requests to the upload/link endpoint result in creating a new dataset.

I added a todo since the above text is not yet implemented.

Reply:
ahh got it! That makes sense now 👌 Thanks for the clarification!